src/lib.rs

```diff
@@ -291,6 +299,76 @@ where
         let writer = self.make_writer.make_writer();
         bufs.flush_current_buf(writer)
     }
+
+    fn get_timestamp<S>(&self, id: &Id, ctx: &Context<S>) -> Option<String>
+    where
+        S: Subscriber + for<'span> LookupSpan<'span>,
+    {
+        match ctx.span(id) {
+            // if the event is in a span, get the span's starting point.
+            Some(ctx) => {
+                let ext = ctx.extensions();
+                let data = ext
+                    .get::<Data>()
+                    .expect("Data cannot be found in extensions");
+
+                if self.config.higher_precision {
+                    Some(self.format_timestamp_with_decimals(data.start))
+                } else {
+                    Some(self.format_timestamp(data.start))
+                }
+            }
+            None => None,
+        }
+    }
+
+    fn format_timestamp(&self, start: std::time::Instant) -> String {
+        let elapsed = start.elapsed();
+        let millis = elapsed.as_millis();
+        let secs = elapsed.as_secs();
+
+        // Convert elapsed time to appropriate units: ms, s, or m.
+        // - Less than 1s : use ms
+        // - Less than 1m : use s
+        // - 1m and above : use m
+        let (n, unit) = if millis < 1000 {
+            (millis as _, "ms")
+        } else if secs < 60 {
+            (secs, "s ")
+        } else {
+            (secs / 60, "m ")
+        };
+
+        let n = format!("{n:>3}");
+        format!(
+            "{timestamp}{unit} ",
+            timestamp = self.styled(Style::new().dimmed(), n),
+            unit = self.styled(Style::new().dimmed(), unit),
+        )
```

Comment:
This is the same in both `format_timestamp*` functions. Maybe move it out of the functions and only return `(n, unit)`

Reply:
Refactored `format_timestamp*` to remove the duplication.